### PR TITLE
Adding Command E2E to PnP

### PR DESCRIPTION
**Pull request by jamdavi**

### CODE REVIEW

File: e2e/test/iothub/command/CommandE2ETests.cs [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+            catch (DeviceNotFoundException ex)
+            {
+                // Although the exception is called "Device" not found, it is used for all 404's, including the 404010 that denotes a module was not found
+                actualErrorCode = ex.Code;
+            }
+
+            Assert.AreEqual(ErrorCode.ModuleNotFound, actualErrorCode);
+
+            await serviceClient.CloseAsync().ConfigureAwait(false);
+        }
```
Comment (jamdavi) [inline e2e/test/iothub/command/CommandE2ETests.cs:194]: Ignore this chunk, leaving these in for when we do the module and service clients.

File: e2e/test/iothub/command/CommandE2ETests.cs [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+
+    public class ServiceCommandRequestObject
+    {
+        public int a { get; set; }
+    }
+
+    public class DeviceCommandResponse
+    {
+        public string name => "e2e_test";
+    }
```
Comment (jamdavi) [inline e2e/test/iothub/command/CommandE2ETests.cs:29]: Added these to handle the serialization scenarios.

File: e2e/test/iothub/command/CommandE2ETests.cs [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+        private async Task SendCommandAndRespondAsync(Client.TransportType transport, Func<DeviceClient, string, MsTestLogger, Task<Task>> setDeviceReceiveMethod, bool withComponent = false, TimeSpan responseTimeout = default, ServiceClientTransportSettings serviceClientTransportSettings = default)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            string commandNameString = CommandName;
+            if (withComponent)
+            {
+                commandNameString = $"{ComponentName}*{CommandName}";
```
Comment (abhipsaMisra) [inline e2e/test/iothub/command/CommandE2ETests.cs:309]: How about we use the pnp specific service client for the service operation; then we wouldn't need to do this string manipulation for command name: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/service/src/DigitalTwin/DigitalTwinClient.cs#L235

File: e2e/test/iothub/command/CommandE2ETests.cs [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+            try
+            {
+                CloudToDeviceMethodResult response =
+                    await serviceClient.InvokeDeviceMethodAsync(
+                        deviceId,
+                        new CloudToDeviceMethod(methodName, methodTimeout).SetPayloadJson(null)).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (!(ex is DeviceNotFoundException))
```
Comment (abhipsaMisra) [inline e2e/test/iothub/command/CommandE2ETests.cs:121]: why are we expecting a `DeviceNotFoundException` here?

Comment (jamdavi) [reply e2e/test/iothub/command/CommandE2ETests.cs]: Copied over from Methods test. Removed

**Pull request closed without merging**